elyra/pipeline/parser.py

```diff
@@ -143,7 +143,8 @@ def _create_pipeline_operation(node: Dict, super_node: Optional[Dict] = None):
             id=node_id,
             type=node.get('type'),
             classifier=node.get('op'),
-            name=PipelineParser._get_ui_data_field(node, 'label'),
+            name=PipelineParser._get_ui_data_field(node, 'label',
+                                                   default_value=PipelineParser._get_app_data_field(node, 'filename')),
```

Comment:
you have filename a few lines below, don't want to move this down below and use that as the default value?

Reply:
I agree.  I guess I was surprised to see the default value added here, but then I noticed that the Operation constructor requires a value for `name` and that the defaulting performed within the `name` property accessor is really all about dropping the suffix.  So, this default value seems fine.

Regarding leveraging the `filename` keyword value as suggested, I'm not sure if that would work.  Can a keyword reference another keyword in a call? No - just checked.  You'll get `NameError: name 'filename' is not defined`.  You'd need to move it into a regular variable prior to the instance construction.